kong.conf.default

```diff
@@ -1447,6 +1447,34 @@
                                  # it might take more time to propagate changes
                                  # to each individual worker.
 
+#router_flavor = traditional_compatible
+                                 # Selects the router implementation to use when
+                                 # performing request routing. Incremental router
+                                 # rebuild is available when the flavor is set
+                                 # to either `expressions` or
+                                 # `traditional_compatible` which could
+                                 # significantly shorten rebuild time for large
+                                 # number of routes.
+                                 #
+                                 # Accepted values are:
+                                 #
+                                 # - `traditional_compatible`: the DSL based expression
+                                 #   router engine will be used under the hood. However
+                                 #   the router config interface will be the same
+                                 #   as `traditional` and expressions are
+                                 #   automatically generated at router build time.
+                                 #   The `expression` field on the `Route` object
+                                 #   is not visible.
+                                 # - `expression`: the DSL based expression router engine
+                                 #   will be used under the hood. Traditional router
+                                 #   config interface is not visible and you must write
+                                 #   Router Expression manually and provide them in the
+                                 #   `expression` field on the `Route` object.
+                                 # - `traditional`: the pre-3.0 Router engine will be
+                                 #   used. Config interface will be the same as
+                                 #   pre-3.0 Kong and the `expression` field on the
+                                 #   `Route` object is not visible.
```

Comment:
Is the traditional mode not recommended and only meant as an escape hatch? Should we document that "traditional" option is deprecated and will be removed in a future release?

Reply:
This has been added, thanks!